server/channels/store/sqlstore/property_field_store.go

```diff
@@ -0,0 +1,339 @@
+		"CreateAt":   field.CreateAt,
+		"UpdateAt":   field.UpdateAt,
+		"DeleteAt":   field.DeleteAt,
+	}, nil
+}
+
+func (s *SqlPropertyFieldStore) propertyFieldToUpdateMap(field *model.PropertyField) (map[string]any, error) {
+	attrsJSON, err := json.Marshal(field.Attrs)
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_to_update_map_marshal_attrs")
+	}
+	if s.IsBinaryParamEnabled() {
+		attrsJSON = AppendBinaryFlag(attrsJSON)
+	}
+
+	return map[string]any{
+		"Name":       field.Name,
+		"Type":       field.Type,
+		"Attrs":      attrsJSON,
+		"TargetID":   field.TargetID,
+		"TargetType": field.TargetType,
+		"UpdateAt":   field.UpdateAt,
+		"DeleteAt":   field.DeleteAt,
+	}, nil
+}
+
+func propertyFieldsFromRows(rows *sql.Rows) ([]*model.PropertyField, error) {
+	results := []*model.PropertyField{}
+
+	for rows.Next() {
+		var field model.PropertyField
+		var attrsJSON string
+
+		err := rows.Scan(
+			&field.ID,
+			&field.GroupID,
+			&field.Name,
+			&field.Type,
+			&attrsJSON,
+			&field.TargetID,
+			&field.TargetType,
+			&field.CreateAt,
+			&field.UpdateAt,
+			&field.DeleteAt,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := json.Unmarshal([]byte(attrsJSON), &field.Attrs); err != nil {
+			return nil, errors.Wrap(err, "property_fields_from_rows_unmarshal_attrs")
+		}
+
+		results = append(results, &field)
+	}
+
+	return results, nil
+}
+
+func propertyFieldFromRows(rows *sql.Rows) (*model.PropertyField, error) {
+	fields, err := propertyFieldsFromRows(rows)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(fields) > 0 {
+		return fields[0], nil
+	}
+
+	return nil, sql.ErrNoRows
+}
+
+type SqlPropertyFieldStore struct {
+	*SqlStore
+}
+
+func newPropertyFieldStore(sqlStore *SqlStore) store.PropertyFieldStore {
+	return &SqlPropertyFieldStore{sqlStore}
+}
+
+func (s *SqlPropertyFieldStore) Create(field *model.PropertyField) (*model.PropertyField, error) {
+	if field.ID != "" {
+		return nil, store.NewErrInvalidInput("PropertyField", "id", field.ID)
+	}
+
+	field.PreSave()
+
+	if err := field.IsValid(); err != nil {
+		return nil, errors.Wrap(err, "property_field_create_isvalid")
+	}
+
+	insertMap, err := s.propertyFieldToInsertMap(field)
+	if err != nil {
+		return nil, err
+	}
+
+	queryString, args, err := s.getQueryBuilder().
+		Insert("PropertyFields").
+		SetMap(insertMap).
+		ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_create_tosql")
+	}
+
+	if _, err := s.GetMaster().Exec(queryString, args...); err != nil {
+		return nil, errors.Wrap(err, "property_field_create_insert")
+	}
+
+	return field, nil
+}
+
+func (s *SqlPropertyFieldStore) Get(id string) (*model.PropertyField, error) {
+	queryString, args, err := s.getQueryBuilder().
+		Select(propertyFieldColumns...).
+		From("PropertyFields").
+		Where(sq.Eq{"id": id}).
+		ToSql()
+	if err != nil {
+		return nil, errors.Wrap(err, "property_field_get_tosql")
+	}
+
+	rows, err := s.GetReplica().Query(queryString, args...)
```

Comment:
Any reason we cannot do:

```
fields := []*model.PropertyField{}
err := s.GetReplica().Select(&fields, queryString, args...)
```
?

Then we don't really need `propertyFieldFromRows` which seems redundant.

If this is about `attrsJSON` and unmarshaling to `field.Attrs`, then we can simply define a custom type with `Value()` and `Scan()` methods for that. sqlx will automatically call those methods to parse out values from and to the DB. See: `func (sa StringArray) Value() (driver.Value, error) {` and `func (sa *StringArray) Scan(value any) error {` in model/utils.go

As a further step, you can even do `s.GetReplica().SelectBuilder()` and pass the builder directly without having to do a `.ToSql` and checking the error output. `(Select/Get)Builder` will do both for you.

Reply:
Yep, the reason was the use of JSON, but I didn't have time to check how this was done during the initial implementation. Will fix in a follow-up PR